### PR TITLE
SONARJAVA-2636 Fix corner case of resolution of conditional operator …

### PR DESCRIPTION
**Pull request by benzonico**
…type

### CODE REVIEW

File: java-frontend/src/test/files/sym/ConditionalOperator.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import java.util.List;
+public class App {
+  public class Foo<F extends List<Object>> {
+  }
+  public class Bar<G extends List<Object>> extends Foo<G> {
+  }
+  private <H extends List<Object>> Foo fun() {
+    return true ? new Bar() : new Foo<List<Object>>(); // resolving type of this expression leads to a failing invariant in LeastUpperBound computation
```
Comment (m-g-sonar) [inline java-frontend/src/test/files/sym/ConditionalOperator.java]: comment should be updated to reflect how it will behave, or at least not tell that it fails.

File: java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java [hunk 2]
```diff
@@ -1602,4 +1601,13 @@ public void fieldType_in_parameterized_hierarchy() throws Exception {
...
     softly.assertAll();
   }
+
+  @Test
+  public void conditional_operator_expression_type() {
+    Result res = Result.createFor("ConditionalOperator");
+    ExpressionTree conditional = ((ReturnStatementTree) ((MethodTree) res.symbol("fun").declaration()).block().body().get(0)).expression();
+    Type conditionalExpressionType = conditional.symbolType();
+    assertThat(conditionalExpressionType.is("App$Foo")).isTrue();
+    assertThat(((JavaType) conditionalExpressionType).isParameterized()).isTrue();
```
Comment (m-g-sonar) [inline java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java]: I would maybe add a test on the parameter type to be complete about the expected type.
